src/network/prover.rs

```diff
@@ -138,9 +154,194 @@ impl<N: Network, E: Environment> Prover<N, E> {
             let _ = handler.await;
         }
 
+        // Initialize the miner, if the node type is a miner.
+        if E::NODE_TYPE == NodeType::Miner && prover.pool.is_none() {
+            Self::start_miner(tasks, prover.clone(), local_ip).await;
+        }
+
+        // Initialize the prover, if the node type is a prover.
+        if E::NODE_TYPE == NodeType::Prover && prover.pool.is_some() {
+            let prover = prover.clone();
+            task::spawn(async move {
```

Comment:
this task needs to be passed to the task handler (`Tasks`) and it would be best if there was an `oneshot` channel that would get notified from within the spawned task that it's been spawned, in order to exit the block only once it's fully ready